### PR TITLE
Bump AppVeyor Visual Studio version to 2022

### PR DESCRIPTION
**Pull request by y-guyon**

### CODE REVIEW

**Review by wantehchang — COMMENTED** [review]
Yannis: Thank you for writing this pull request. Do you think Visual Studio 2022 is likely to fix the build timeout? Or are you just trying to use the latest version of Visual Studio?

File: appveyor.yml [hunk 1]
```diff
@@ -17,7 +17,7 @@ install:
  # Get submodules
  - git submodule update --init --recursive
  # Prepare VS env
- - call "C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Auxiliary\Build\vcvars64.bat"
+ - call "C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Auxiliary\Build\vcvarsall.bat" x64 8.1
```
[the remark this answers is not in the document]
Comment (wantehchang) [reply appveyor.yml:20]: Also, the Windows 8.1 SDK is very old. It seems that we should use at least a Windows10 SDK.

[the remark this answers is not in the document]
Comment (wantehchang) [reply appveyor.yml:20]: To investigate the following compilation error:
```
C:\projects\libavif-93od3\ext\aom\aom_ports/x86.h(151): fatal error C1083: Cannot open include file: 'windows.h': No such file or directory
```

we can add a command to print the value of the `INCLUDE` environment variable after preparing VS env:

```
set INCLUDE
```

or

```
echo %INCLUDE%
```

**Comment by y-guyon** [comment]
> Yannis: Thank you for writing this pull request. Do you think Visual Studio 2022 is likely to fix the build timeout? Or are you just trying to use the latest version of Visual Studio?

@jzern mentioned that the build took only 17 minutes on a local machine with a more recent Visual Studio version than 2019, so I thought I would give it a try.

> Also, the Windows 8.1 SDK is very old. It seems that we should use at least a Windows10 SDK.

I agree but Windows 10 SDK generated [meson's "Could not find Windows resource compiler"](https://ci.appveyor.com/project/louquillio/libavif/builds/44295792) so I tried 8.1.

> `call "C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Auxiliary\Build\vcvars64.bat"`

Good catch, a Google search about "missing vcvars64.bat VS 2022" suggested `vcvarsall.bat`, because this is `Program Files` and not `Program Files (x86)`. Anyway this seems to work, but [the 1h timeout is still here](https://ci.appveyor.com/project/louquillio/libavif/builds/44303372).

**Comment by wantehchang** [comment]
Thanks for the reply. We can speed up the AppVeyor build by building less code. I am going to try that approach in https://github.com/AOMediaCodec/libavif/pull/1000.

The build is timing out in rav1e.cmd, so I am removing that first. In the worst case, we only need libaom as the AV1 encoder and dav1d as the AV1 decoder.

Also, if the AppVeyor build is intended to provide binary releases for Windows, then it should build and use libyuv. I can do that in https://github.com/AOMediaCodec/libavif/pull/1000.

**Review by wantehchang — APPROVED** [review]
LGTM. Please merge this after https://github.com/AOMediaCodec/libavif/pull/1000 is merged and **wait for a successful AppVeyor build**.

**Review by wantehchang — CHANGES_REQUESTED** [review]
Could you also change "Visual Studio 2019" to "Visual Studio 2022" in `README.md`? Thanks.

**Comment by wantehchang** [comment]
I have merged https://github.com/AOMediaCodec/libavif/pull/1000.

**Review by wantehchang — APPROVED** [review]
LGTM. AppVeyor says the build completed in 40 min 18 sec.

Please use the "Squash and merge" method to merge this pull request. Thanks.

**Pull request merged**